### ISSUE TITLE
Missing require for stdlib class Set

### ISSUE THREAD

**Issue by sbrinkhorst** [davishmcclurg/json_schemer#98]
We got this error in our gitlab pipeline:
```
Failure/Error: require 'json_schemer'
NameError:
  uninitialized constant JSONSchemer::Schema::Base::Set
# /home/gitlab-runner/.gem/ruby/2.6.0/gems/json_schemer-0.2.18/lib/json_schemer/schema/base.rb:24:in `<class:Base>'
# /home/gitlab-runner/.gem/ruby/2.6.0/gems/json_schemer-0.2.18/lib/json_schemer/schema/base.rb:4:in `<module:Schema>'
# /home/gitlab-runner/.gem/ruby/2.6.0/gems/json_schemer-0.2.18/lib/json_schemer/schema/base.rb:3:in `<module:JSONSchemer>'
# /home/gitlab-runner/.gem/ruby/2.6.0/gems/json_schemer-0.2.18/lib/json_schemer/schema/base.rb:2:in `<top (required)>'
# /home/gitlab-runner/.gem/ruby/2.6.0/gems/json_schemer-0.2.18/lib/json_schemer.rb:19:in `require'
# /home/gitlab-runner/.gem/ruby/2.6.0/gems/json_schemer-0.2.18/lib/json_schemer.rb:19:in `<top (required)>'
```
We're unsure why it suddenly became a problem, it worked for a long time and still works locally. The solution is to put `require 'set'` in the right place, which seems to be in `base.rb`, because `Set` is only used in that class and it's derivatives.

**Comment by AJFunk** [comment]
Caused by upgrading regexp_parser to 2.1.1 https://github.com/davishmcclurg/json_schemer/commit/ab5ea07e21633952f5025aaabcfbb37c05c3a74d#diff-89cade48462044ee1b672dc5f4c3ec250fbd29effcd8932096a23c1283c6731fR18

https://github.com/ammar/regexp_parser/commit/49420efc614aebc9d729c15a927a2288b048fb9c

**Comment by davishmcclurg** [comment]
> Caused by upgrading regexp_parser to 2.1.1 [ab5ea07#diff-89cade48462044ee1b672dc5f4c3ec250fbd29effcd8932096a23c1283c6731fR18](https://github.com/davishmcclurg/json_schemer/commit/ab5ea07e21633952f5025aaabcfbb37c05c3a74d#diff-89cade48462044ee1b672dc5f4c3ec250fbd29effcd8932096a23c1283c6731fR18)
> 
> [ammar/regexp_parser@49420ef](https://github.com/ammar/regexp_parser/commit/49420efc614aebc9d729c15a927a2288b048fb9c)

Good find. Interesting that they decided to remove set, since it's in the standard library.

Thanks for opening the issue, @sbrinkhorst. One small change on the PR, otherwise looks good.

**Comment by davishmcclurg** [comment]
Fixed in 0.2.19. Thanks!
